src/psyclone/generator.py

```diff
@@ -98,29 +98,35 @@ def handle_script(script_name, info, function_name, is_optional=False):
 
     '''
     # pylint: disable=too-many-locals
-    sys_path_appended = False
+    sys_path_prepended = False
     try:
-        # a script file has been provided, check it exists
-        if not os.path.isfile(script_name) and not any(
-               os.path.isfile(os.path.join(p, script_name)) for p in sys.path):
-            raise GenerationError(
-                f"generator: script file '{script_name}' not found")
         filepath, filename = os.path.split(script_name)
-        if filepath:
-            # a path has been provided, so add it to the python search path
-            sys_path_appended = True
-            sys.path.append(filepath)
-        filename, fileext = os.path.splitext(filename)
+        module_name, fileext = os.path.splitext(filename)
+        # the file must either be:
+        # a) at the given path or, given no path, in the current directory; or
+        # b) given no path, in the system path
+        if not (os.path.isfile(script_name) or
+                not filepath and any (os.path.isfile(os.path.join(p, filename))
+                                      for p in sys.path)):
+                raise GenerationError(
```

Comment:
pylint doesn't like the indentation here. (e.g. you need to indent the above two lines an extra 4 spaces).

Reply:
Done.